netket/exact.py

```diff
@@ -25,3 +65,134 @@ def _ImagTimePropagation_iter(self, dt, n_iter=None):
 
 
 ImagTimePropagation.iter = _ImagTimePropagation_iter
+
+
+@deprecated()
+class EdResult(object):
+    def __init__(self, eigenvalues, eigenvectors):
+        # NOTE: These conversions are required because our old C++ code stored
+        # eigenvalues and eigenvectors as Python lists :(
+        self._eigenvalues = eigenvalues.tolist()
+        self._eigenvectors = (
+            [eigenvectors[:, i] for i in range(eigenvectors.shape[1])]
+            if eigenvectors is not None
+            else []
+        )
+
+    @property
+    def eigenvalues(self):
+        r"""Eigenvalues of the Hamiltonian.
+        """
+        return self._eigenvalues
+
+    @property
+    def eigenvectors(self):
+        r"""Eigenvectors of the Hamiltonian.
+        """
+        return self._eigenvectors
+
+    def mean(self, operator, which):
+        import numpy
+
+        x = self._eigenvectors[which]
+        return numpy.vdot(x, operator(x))
+
+
+@deprecated(
+    "use 'netket.Operator.to_linear_operator' to convert an operator "
+    "to a 'scipy.sparse.linalg.LinearOperator' and then call "
+    "'scipy.sparse.linalg.eigsh' directly"
+)
+def lanczos_ed(
+    operator,
+    matrix_free=False,
+    first_n=1,
+    max_iter=1000,
+    seed=None,
+    precision=1e-14,
+    compute_eigenvectors=False,
+):
+    r"""Uses Lanczos algorithm to diagonalize the operator.
```

Comment:
Strictly speaking, computing the first `n` eigenvalues (or even eigenvectors) does not diagonalize the operator. Only computing the full basis would. So maybe this comment should be more specific (e.g., `Computes the first n eigenvalues and, optionally, eigenvectors of a Hermitian operator using the Lanczos method.`).